### ISSUE TITLE
Implement periodic offshore DB backup on production DB

### ISSUE THREAD

**Issue by mhetreramesh** [phpreboot/website#70]

**Comment by kapilsharma** [comment]
I see two potential problems. Let's discuss before you start work on it.

- Finance. Total annual expenses of PHP reboot is approx 20-25 K. Without a sponsor, I don't want to use any further paid service, unless necessary.
- We already have total DB backup system in form of JSON. Obviously, it does not include users but right now we do not have user-specific features.

Still it will be needed in future but for now, I'd prefer to keep it as low priority and in backlog. Please suggest.

**Comment by mhetreramesh** [comment]
We can use backblaze cloud storage, it provides 10 gb storage for free.
Also we can use https://github.com/spatie/laravel-backup package, its really easy to implement.
I have created a package for backblaze for php flysystem https://packagist.org/packages/mhetreramesh/flysystem-backblaze

So at the end I would say, its really easy and financially cheaper job to do. You can review things and then we will decide.

**Comment by kapilsharma** [comment]
Ahh I just saw personal storage pricing, not cloud storage.

Make sense. Go ahead.

Just one point. I went through code on mhetreramesh/flysystem-backblaze. Many functions still having `echo` & `die` statements. Please complete them before using,

Please think a bit on how can we keep it vendor free so that in future, if needed, we can easily migrate to another provider. How are you planning to do it? Will you create artisan command or scheduler or something else?

**Comment by mhetreramesh** [comment]
Yes, I'm working on that repository also. I'm going to make it proper and higher. 
Yes this backblaze package I'm writing is a vendor free, actually its an adapter for php flysystem and is pretty much generic. There is not any type of logic change practically to change the vendor in future.
One request, can you please create an backblaze/dropbox account? This is not actually a major task, so I think we can wrap it up quickly.

**Comment by kapilsharma** [comment]
Please also check
https://docs.spatie.be/laravel-backup/v4/introduction

Will it help? I just read article while adding to magazine, didn't tried out.

**Comment by mhetreramesh** [comment]
Yes, W'll be going to use the same package for matching version of course!

**Comment by mhetreramesh** [comment]
Started working on this
